### ISSUE TITLE
Modify pdo-shell to enable asynchronous commits

### ISSUE THREAD

**Issue by prakashngit** [hyperledger-labs/private-data-objects#216]
While the new commit modules are integrated into the PDO shell, the shell commits synchronously. Modify the shell to enable asynchronous commits. The following series of steps is one way to accomplish this:

1. make wait_for_commit a separate shell command. 
2. Add a provision to specify commit dependencies as input to send command.
3. Return commit_id after send command back to shell so as to use as input commit_dependency in a future send command

**Comment by cmickeyb** [comment]
I don't think we need a separate command. We need to handle the "wait" flag correctly. That way, any operation that updates state can wait for the commit. If the dependencies are handled correctly (and they should be) then that will wait until all dependent transactions commit. 

I think a more interesting question is whether we should provide a timeout for the wait parameter for the operations that are performing a send. Having a timeout feels like a better way to handle failure to commit.
